### PR TITLE
EZP-30019: Revise configuration doc

### PR DESCRIPTION
**Pull request by DominikaK**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | https://jira.ez.no/browse/EZP-30019
| Versions      | all

Note: Section about `default_page` removed following https://github.com/ezsystems/ezplatform-admin-ui/pull/378

Note 2: for reviews, look at commit https://github.com/ezsystems/developer-documentation/commit/ff496647fa423fa4dd33b2db6a00ed06ed43a6bd. The first commit moves text around and GitHub isn't good at presenting such changes.

### CODE REVIEW

File: docs/guide/config_dynamic.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+    public function __construct( array $languages )
+    {
+        $this->languages = $languages;
+    }
+}
+```
+
+!!! tip
+
+    Setter injection for dynamic settings should always be preferred, as it makes it possible to update your services that depend on them when ConfigResolver is updating its scope (e.g. when previewing content in a given SiteAccess). **However, only one dynamic setting should be injected by setter**.
```
Comment (MagdalenaZuba) [inline docs/guide/config_dynamic.md]: Setter injection for dynamic settings should always **be preferred method. It allows you** to update your services that **depend on it** when ConfigResolver is updating its scope (e.g. when previewing content in a given SiteAccess).

File: docs/guide/config_repository.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+``` yaml
+repositories:
+    default:
+        fields_groups:
+            list: [content, features, metadata]
+            default: content
+```
+
+These identifiers can be given human-readable values and translated. Those values are used when editing Content Types.
```
Comment (MagdalenaZuba) [inline docs/guide/config_repository.md]: values and translations.

File: docs/guide/configuration.md [hunk 3]
```diff
@@ -18,9 +18,12 @@ Main configuration files are located in the `app/config` folder.
...
 - `routing.yml` defines routes that will be used throughout the application.
 
-Configuration can be made environment-specific using separate files for each environment. These files contain additional settings and point to the general (not environment-specific) configuration that is applied in other cases.
+Configuration can be made environment-specific using separate files for each environment.
+These files contain additional settings and point to the general (not environment-specific) configuration that is applied in other cases.
 
-Here you can read more about [how configuration is handled in Symfony](http://symfony.com/doc/current/best_practices/configuration.html).
+!!! tip
+
+    Read more about [how configuration is handled in Symfony](http://symfony.com/doc/current/best_practices/configuration.html).
```
Comment (MagdalenaZuba) [inline docs/guide/configuration.md]: should we link to current or 3.4? - question to all symfony links

Comment (konradoboza) [reply docs/guide/configuration.md]: We definitely should.

**Comment by DominikaK** [comment]
> As far as I understand, you removed `Removing old versions` section as it isn't explicitly related to configuration? Besides, I would say it is cleaner now, so +1.

Actually it's still there: https://github.com/ezsystems/developer-documentation/pull/509/files#diff-c9de29e219e5e08f7724b6e75881b89eR124

**Pull request merged**